packages/x-tree-view/src/TreeItem/useTreeItemState.ts

```diff
@@ -96,6 +97,7 @@ export function useTreeItemState(itemId: string) {
     handleExpansion,
     handleSelection,
     handleCheckboxSelection,
+    handleContentClick: onItemClick,
```

Comment:
This hook will be removed in v8 since it's only used on the legacy component
So I just returned `onItemClick` to give a way for people using `ContentComponent` to handle the new prop, but I don't expect it to be widely used.